projects/audit-logging.md

```diff
@@ -0,0 +1,196 @@
+- REQ-PIPE-01: OTel collector must provide guaranteed delivery of audit logs, including when its process is interrupted
+
+See [Appendix A: Guarantee of Delivery] for more details
+
+## Deliverables
+
+- semantic convention for audit logs
+- extend OTel APIs/SDKs for audit logging purposes (in collaboration with the respective SIG)
+- extend OTel collector for audit logging purposes (in collaboration with the respective SIG)
+
+## Staffing / Help Wanted
+
+The following vendors are interested in improving this area:
+- SAP (@mlenkeit, @FWinkler79)
+- Microsoft (@reyang)
+
+Other vendors are invited to join the discussion.
+
+### Required staffing
+
+* Project lead: SAP (name tbd)
+* Sponsors:
+  - @reyang
+  - tbd
+* GC liaison: @svrnm
+* Engineers:
+  * SAP will provide a prototype in two languages (tbd; likely two of Java, JavaScript, Go)
+* Maintainers/approvers: tbd
+
+## Timeline
+
+TBD based on community involvement.
+
+## Labels
+
+- audit-logging (tbc)
+
+## Project Board
+
+TODO: add link
+
+## SIG Meetings and Other Info
+
+TODO: add information
+
+## Appendix
+
+### Appendix A: Guarantee of Delivery
+
+In the context of this document, guarantee of delivery describes the ability of delivering audit logs from source to destination through OTel means while ensuring that all such signals arrive at the destination and/or providing the source with a means to handle failed delivery.
+
+Messaging protocols that support different levels of delivery guarantees may refer to this behavior as _at least once_ or _exactly once_, as opposed to _at most once_.
+
+We assume that every component that is involved in the delivery of audit logs from source to destination must support guarantee of delivery individually, rather than assuming that this ability can be provided by e.g. only the collector or SDK.
+
+The implications of guarantee of delivery can be illustrated with an example consisting of a workload, an OTel collector, and a durable storage. The workload acts as the source and produces audit logs via the OTel SDK/API. It writes the data via OTLP to the OTel collector. The OTel collector is configured to export audit logs to a durable storage that acts as the destination such as an S3 bucket.
+
+The following implications would apply:
+
+- workload produces an audit-relevant event:
+
+  The workload emits the event via the OTel SDK/API. It may wait for acknowledgement of receipt from the collector before proceeding. If the event is rejected or receipt is not acknowledged in time, the workload or SDK may act accordingly, e.g. retry, rollback a database transaction, inform the user, etc.
+
+- OTel collector receives the event:
+
+  To ensure that the event is not lost even if the collector process is terminated or crashes, the collector may need to persist the event before acknowledging receipt to the workload or SDK. If the event cannot be persisted, receipt must be rejected.
```

Comment:
What is the expectation if the collector instance disappeared (e.g., the machine running the collector exploded / was stolen)?

Reply:
I think this is the most tricky part, or to put it in a question: do we need guarantee of delivery between 2 components (workload->collector,collector->S3) or end-to-end (workload->S3)?

I would assume "end-to-end" except the collector can guarantee that data is persisted according the auditing requirements

Reply:
> (e.g., the machine running the collector exploded / was stolen)
_If_ the solution for audit logging with OTel meant that the OTel Collector had an own persistence, I would argue that theft/explosion/etc. are rather in the responsibility of Operations in terms of configuring said persistence such that it is resilient "enough".
Or to make this more concrete: _if_ for example something such as the [storage extension](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/storage) was used, Operations would need to make sure that the database/file/redis storage runs in an HA mode.
I'm stressing the _if_ here, because I think is a detail that the SIG should work out. Or do you think that's something that should rather be clarified upfront?

Reply:
> I'm stressing the _if_ here, because I think is a detail that the SIG should work out. Or do you think that's something that should rather be clarified upfront?

I suggest that we leave this for the SIG to figure out. In the OTEP, I suggest that we avoid "guaranteed delivery" and use something like "certain degree/level of data delivery guarantee". Not a blocker for this PR though (I'm good with the current version).

Reply:
+1 for what @reyang wrote. I think it is good to have this in the appendix and some wording around it, since there is many people (including myself) who have a superficial knowledge around audit logs, so it helps to contextualize and understand what this is all about